packages/ui-components/src/lib/form/Form.tsx

```diff
@@ -25,24 +26,56 @@ export interface UnsavedChangesConfig {
   modalProps?: UnsavedChangesModalProps;
 }
 
-export interface FormProps<T> extends FormikConfig<T> {
+export interface PersistValuesConfig<T> {
+  persistFunction: (values: T) => void;
+  debounce?: number;
+}
+
+export interface FormProps<T> extends Omit<FormikConfig<T>, 'onSubmit'> {
   className?: string;
   confirmUnsavedChanges?: boolean;
+  persistValuesConfig?: PersistValuesConfig<T>;
   unsavedChangesConfig?: UnsavedChangesConfig;
   withoutFormElement?: boolean;
   formConfig?: FormConfig;
   children: (formikProps: FormikProps<T>) => ReactElement;
+  onSubmit?: (values: T, formikHelpers: FormikHelpers<T>) => void | Promise<any>;
+}
+
+interface PersistedValuesProps<T> extends PersistValuesConfig<T> {
+  formikProps: FormikProps<T>;
 }
 
-const FormContent: FC<{
+function PersistedValues<T>(props: PersistedValuesProps<T>) {
+  const { formikProps, persistFunction, debounce } = props;
+  const { values } = formikProps;
+
+  usePersistedFormValues(values, persistFunction, debounce);
+
+  return null;
+}
```

Comment:
Could we move this to its own component file, along with its `interface`?

Reply:
@jaredhill4 do we need to? all of these components are tied together. https://twitter.com/dan_abramov/status/1220029098543788032?s=20

Reply:
🤷 It could technically be used as a standalone component, right? It just seems consistent for standalone components to keep them in their own file. By that logic, you could argue that none of our components should be in separate files, or that all component variants should be in the same file (i.e., buttons, inputs, etc.).